### PR TITLE
[#3681] Fixed iscan return codes and error messages

### PR DESCRIPTION
**Pull request by andrew-irods**

### CODE REVIEW

File: lib/core/src/scanUtil.cpp [hunk 1]
```diff
@@ -87,9 +76,13 @@ scanObjDir( rcComm_t *conn, rodsArguments_t *myRodsArgs, const char *inpPath, co
...
     }
 
+    // This variable will contain either the 0, or the last error
+    // encountered as the loop below iterates through all of the
+    // entries in the physical directory.
+    int return_status = 0;
     directory_iterator end_itr; // default construction yields past-the-end
     for ( directory_iterator itr( srcDirPath ); itr != end_itr; ++itr ) {
-        path cp = itr->path();
+    	path cp = itr->path();
```
Comment (trel) [inline lib/core/src/scanUtil.cpp]: need a space

File: lib/core/src/scanUtil.cpp [hunk 2]
```diff
@@ -104,9 +97,11 @@ scanObjDir( rcComm_t *conn, rodsArguments_t *myRodsArgs, const char *inpPath, co
         else {
             status = chkObjExist( conn, fullPath, hostname );
         }
+        if (status != 0) {
+        	return_status = status;
```
Comment (trel) [inline lib/core/src/scanUtil.cpp]: just four spaces please

File: scripts/irods/test/test_iscan.py [hunk 3]
```diff
@@ -70,3 +77,102 @@ def test_iscan_data_object(self):
...
+        self.admin.assert_icommand('ireg -R {0} -C {1} {2}/{3}'.format(self.testresc, os.path.abspath(self.dirname2),
+                                                                self.admin.session_collection,
+                                                                self.dirname2))
+
+        # At this point, we have 2 files in each dir, one called "0",
+        # which is 0 length, and the other called "1", which is not 0 length.
+
+        # First verify correct behavior with no missing files on the filesystem:
+        self._util_simple_icmd_assert('iscan -r {0}'.format(os.path.abspath(self.dirname1)))
+    	self._util_simple_icmd_assert('iscan -r {0}'.format(os.path.abspath(self.dirname2)))
```
Comment (trel) [inline scripts/irods/test/test_iscan.py]: surprised this runs - tabs to whitespace here to line 132

**Comment by andrew-irods** [comment]
Continuous Integration has not been run yet.

**Comment by andrew-irods** [comment]
Jenkins job #761 is currently running CI:  http://172.25.14.125:8080/view/2.%20Personal/job/irods-build-and-test-workflow/761/

**Pull request merged**